.github/workflows/post_merge_integration.yml

```diff
@@ -42,75 +39,122 @@ jobs:
+          key: "poetry-installer-1.2.1"
+      - name: Download poetry installer
+        if: steps.cache-poetry-installer.outputs.cache-hit != 'true'
         run: |
-          sudo apt-get update --fix-missing && sudo apt-get -y install libgdal-dev wkhtmltopdf
-      - name: Install gdal python (api)
-        # We don't have much control over what version of gdal we're getting, it's pretty much whatever is
-        # available to us. As such, gdal is not installed by poetry, since the version will differ between
-        # platforms.
-        # Also have to be absolutely sure that numpy is installed before gdal (otherwise gdal doesn't install
-        # correctly, and you'll run into ndarray errors.)
-        # On ubuntu20.04, the most recent we have is gdal, so we use pygdal to install it. (used to use
-        # gdal, but it now has a bug that causes it to fail to install on ubuntu20.04)
-        working-directory: ./api
+          echo
+          mkdir ~/poetry_installer
+          curl -sSL https://install.python-poetry.org > ~/poetry_installer/install-poetry.py
+      - name: Install poetry (api)
         run: |
-          poetry run pip install numpy
-          poetry run pip install pygdal==3.0.4.10
-      - name: Lint (api)
-        # We used to be able to do linting before installing gdal, but it's not possible anymore.
-        # We can however place it ahead of the R installs.
+          cd ~/poetry_installer
+          python install-poetry.py --version 1.2.1
+          poetry config virtualenvs.create true
+          poetry config virtualenvs.in-project false
+      # poetry cache folder: /home/runner/.cache/pypoetry
+      - name: Cache poetry
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-poetry
+        with:
+          path: ~/.cache/pypoetry
+          key: ${{ runner.os }}-poetry-1.2.1-cache-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-poetry-1.2.1-cache-
+      - name: Cache .venv
+        id: cache-venv
+        uses: actions/cache@v3
+        with:
+          path: ~/work/wps/wps/api/.venv
+          key: ${{ runner.os }}-venv-poetry-1.2.1-${{ hashFiles('**/poetry.lock') }}
+      - name: Install python dependencies using poetry (api)
+        if: steps.cache-venv.outputs.cache-hit != 'true'
         working-directory: ./api
         run: |
-          poetry run pylint --rcfile=.pylintrc app/*.py app/**/*.py
+          poetry run python -m pip install --upgrade pip
+          poetry install
+          poetry run python -m pip install gdal==$(gdal-config --version)
       - uses: r-lib/actions/setup-r@v2
         with:
           r-version: ${{ matrix.R }}
-      - name: Cache renv (api)
-        # You have to create the renv.lock file on your local machine
-        # install.packages("renv")
-        # renv::install("cffdrs", dependencies = TRUE)
-        # renv::snapshot()
+      - name: Cache /home/runner/work/_temp/Library
```

Comment:
It's more kinda 4gl. So you specify the cache first, so that it's loaded if it's there. If it's not there, the workflow remembers that you wanted a cache - and does that at the end. If you look at the some of the workflows that have run, you'll see it does the actual storing of the cache once everything is done.